packages/hardhat-core/test/internal/artifacts.ts

```diff
@@ -909,6 +910,213 @@ describe("Artifacts class", function () {
+
+    const SOLC_INPUT: CompilerInput = {
+      sources: {},
+      settings: { optimizer: {}, outputSelection: {} },
+      language: "",
+    };
+
+    const SOLC_OUPUT: CompilerOutput = {
+      sources: {},
+      contracts: {},
+    };
+
+    let artifacts: Artifacts;
+    let buildInfoPath: string;
+    beforeEach(async function () {
+      artifacts = new Artifacts(this.tmpDir);
+      buildInfoPath = await artifacts.saveBuildInfo(
+        "0.4.12",
+        "0.4.12+123",
+        SOLC_INPUT,
+        SOLC_OUPUT
+      );
+
+      await artifacts.saveArtifactAndDebugFile(
+        {
+          _format: "",
+          abi: [],
+          contractName: "C",
+          sourceName: "c.sol",
+          bytecode: "",
+          deployedBytecode: "",
+          linkReferences: {},
+          deployedLinkReferences: {},
+        },
+        buildInfoPath
+      );
+    });
+
+    describe("Successful caching", function () {
+      it("Should cache the path to each artifact and build info, and the lists of paths", async function () {
+        assert.lengthOf(await artifacts.getArtifactPaths(), 1);
+        assert.lengthOf(await artifacts.getDebugFilePaths(), 1);
+        assert.lengthOf(await artifacts.getBuildInfoPaths(), 1);
+
+        // we create some other files, which shouldn't be returned
+
+        await fsExtra.writeJSON(path.join(this.tmpDir, "c.sol", "B.json"), {});
+        await fsExtra.writeJSON(
+          path.join(this.tmpDir, "c.sol", "B.dbg.json"),
+          {}
+        );
+        await fsExtra.writeJSON(
+          path.join(this.tmpDir, "build-info", "something.json"),
+          {}
+        );
+
+        assert.lengthOf(await artifacts.getArtifactPaths(), 1);
+        assert.lengthOf(await artifacts.getDebugFilePaths(), 1);
+        assert.lengthOf(await artifacts.getBuildInfoPaths(), 1);
+      });
+    });
+
+    describe("Clear cache", function () {
+      it("Should clear all the caches", async function () {
+        assert.lengthOf(await artifacts.getArtifactPaths(), 1);
+        assert.lengthOf(await artifacts.getDebugFilePaths(), 1);
+        assert.lengthOf(await artifacts.getBuildInfoPaths(), 1);
+
+        // we create some other files, which should be returned after clearin
+        //  the cache
+
+        await fsExtra.writeJSON(path.join(this.tmpDir, "c.sol", "B.json"), {});
+        await fsExtra.writeJSON(
+          path.join(this.tmpDir, "c.sol", "B.dbg.json"),
+          {}
+        );
+        await fsExtra.writeJSON(
+          path.join(this.tmpDir, "build-info", "something.json"),
+          {}
+        );
+
+        artifacts.clearCache();
+
+        assert.lengthOf(await artifacts.getArtifactPaths(), 2);
+        assert.lengthOf(await artifacts.getDebugFilePaths(), 2);
+        assert.lengthOf(await artifacts.getBuildInfoPaths(), 2);
+      });
+
+      it("Shouldn't re-enable the cache", async function () {
+        artifacts.disableCaching();
+        artifacts.clearCache();
```

Comment:
I guess `disableCaching` should call `clearCache`?

Reply:
It was already working like that. This test was wrong, so I fixed it, and also added a test verifying that `disableCaching` actually clears it.